pkg/rancher-desktop/config/settings.ts

```diff
@@ -427,3 +430,114 @@ function updateSettings(settings: Settings) {
 // Imported from dashboard/config/settings.js
 // Setting IDs
 export const SETTING = { PL_RANCHER_VALUE: 'rancher' };
+
+/**
+ * Read and validate the json formatted deployment profile.
+ * Returns the system profile if it exists, otherwise the user profile if it exists.
+ * @returns A settings profile stripped of all fields not found in the schema.
+ */
+export function readDeploymentProfile() {
+  let profileData;
+  let profile;
+
+  switch (os.platform()) {
+  case 'win32':
+    try {
+      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
+    } catch {
+      try {
+        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
+      } catch {}
+    }
+    break;
+  case 'linux':
+    try {
+      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_JSON), 'utf8');
+    } catch {
+      try {
+        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_JSON), 'utf8');
+      } catch {}
+    }
+    if (typeof profileData !== 'undefined') {
+      profile = JSON.parse(profileData);
+    }
+    break;
+  case 'darwin':
+    try {
+      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_PLIST), 'utf8');
+    } catch {
+      try {
+        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_PLIST), 'utf8');
+      } catch {}
+    }
+    if (typeof profileData !== 'undefined') {
+      profile = plist.parse(profileData);
+    }
+    break;
+  }
+
+  return validateDeploymentProfile(profile, defaultSettings);
+}
+
+function readRegistryDeploymentProfile() {
+  /* Not implemented */
```

Comment:
I would throw an exception here, like 
```
throw new Error('INTERNAL ERROR: readRegistryDeploymentProfile not yet implemented');
```

Reply:
The exception is a better idea.  changed.